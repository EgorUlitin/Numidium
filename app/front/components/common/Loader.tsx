import * as React from 'react'

import { Spin } from 'antd'
import { css } from 'emotion'


interface Props {
    loading: boolean
    error: boolean
    children?: any
}

export default (props: Props) =>
    props.error
        ? <p>error</p>
        : props.loading
            ? <div className={s.spinContainer}><Spin /></div>
            : props.children

const s = {
    spinContainer: css`
        text-align: center;
    `,
}
