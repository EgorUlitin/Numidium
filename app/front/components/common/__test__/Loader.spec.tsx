import * as React from 'react'
import * as renderer from 'react-test-renderer'

import Loader from '../Loader'

test('`Loader` renders correctly with loading state', () => {
    const tree = renderer
        .create(
            <Loader loading={true} error={false}>
                <p>Content</p>
            </Loader>,
        )
        .toJSON()

    expect(tree).toMatchSnapshot()
})

test('`Loader` renders correctly with error state', () => {
    const tree = renderer
        .create(
            <Loader loading={false} error={true}>
                <p>Content</p>
            </Loader>,
        )
        .toJSON()

    expect(tree).toMatchSnapshot()
})

test('`Loader` renders correctly with content', () => {
    const tree = renderer
        .create(
            <Loader loading={false} error={false}>
                <p>Content</p>
            </Loader>,
        )
        .toJSON()

    expect(tree).toMatchSnapshot()
})
