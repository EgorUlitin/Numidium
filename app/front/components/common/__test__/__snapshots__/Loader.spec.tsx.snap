// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`Loader\` renders correctly with content 1`] = `
<p>
  Content
</p>
`;

exports[`\`Loader\` renders correctly with error state 1`] = `
<div
  className="ant-alert ant-alert-error ant-alert-no-icon"
  data-show={true}
  style={undefined}
>
  <span
    className="ant-alert-message"
  >
    Произошла ошибка, попробуйте обновить страницу
  </span>
  <span
    className="ant-alert-description"
  />
</div>
`;

exports[`\`Loader\` renders correctly with loading state 1`] = `
<div
  className="css-gng0k7"
>
  <div
    className="ant-spin ant-spin-spinning ant-spin-show-text"
  >
    <span
      className="ant-spin-dot ant-spin-dot-spin"
    >
      <i />
      <i />
      <i />
      <i />
    </span>
  </div>
</div>
`;
